pkg/proxy/iptables/proxier.go

```diff
@@ -584,15 +584,28 @@ func buildEndpointInfoList(endPoints []hostPortInfo, endpointIPs []string) []*en
 	return filteredEndpoints
 }
 
-type healthCheckEndpoints struct {
-	name      types.NamespacedName
-	endpoints []hostPortInfo
+// Map of namespaced names (eg, service namespace/name) of a service to either
+// an empty set, or a set container exactly the same namespaced service name as
+// a string when the service has local endpoints.  We need this later because
+// healthcheck.UpdateEndpoints() is a silly API.
```

Comment:
Heh. I didn't mean to literally use that text, although I suppose it's true...
(s/container/containing/)